app/controllers/admin/teachers_controller.rb

```diff
@@ -42,6 +42,35 @@ class TeachersController < Admin::ApplicationController
     #     transform_values { |value| value == "" ? nil : value }
     # end
 
+    # need to create the new teacher by the admin
+    def create
+      temp_password = Devise.friendly_token.first(20)
+      @teacher = Teacher.new(teacher_params.merge(password: temp_password))
+      if @teacher.save
+        # checking if any classroom_id is provided or not
+        classroom_id = params[:teacher][:classroom_id]
```

Comment:
Great work on this feature! 🎉

  The admin teacher creation functionality is well-implemented with good security practices and error handling.

  Just one small item: please add classroom_id to the params.expect in teacher_params for consistency with the other parameters.

  Thanks for including tests!

  Solid work! Polly asked me to look at this